tests/integration_tests/performance/test_network_tcp_throughput.py

```diff
@@ -228,78 +242,75 @@ def consume_iperf_tcp_output(cons, result, vcpus_count):
+            "--len", f"{payload_length}"
         )
 
-        if payload_length != "DEFAULT":
-            iperf_guest_cmd_builder = iperf_guest_cmd_builder.with_arg(
-                "--len", f"{payload_length}"
-            )
-
-        iperf3_id = f"tcp-p{payload_length}-wsDEFAULT-{mode}"
-
-        cons = consumer.LambdaConsumer(
-            metadata_provider=DictMetadataProvider(
-                measurements=CONFIG_DICT["measurements"],
-                baseline_provider=NetTCPThroughputBaselineProvider(env_id, iperf3_id),
-            ),
-            func=consume_iperf_tcp_output,
-            func_kwargs={"vcpus_count": basevm.vcpus_count},
-        )
-
-        prod_kwargs = {
-            "guest_cmd_builder": iperf_guest_cmd_builder,
-            "basevm": basevm,
-            "current_avail_cpu": current_avail_cpu,
-            "runtime": CONFIG_DICT["time"],
-            "omit": protocol["omit"],
-            "load_factor": CONFIG_DICT["load_factor"],
-            "modes": CONFIG_DICT["modes"][mode],
-        }
-        prod = producer.LambdaProducer(produce_iperf_output, prod_kwargs)
-        yield cons, prod, f"{env_id}/{iperf3_id}"
-
-
-def pipes(basevm, host_ip, current_avail_cpu, env_id):
-    """Pipes generator."""
-    for mode in CONFIG_DICT["modes"]:
-        # We run bi-directional tests only on uVM with more than 2 vCPus
-        # because we need to pin one iperf3/direction per vCPU, and since we
-        # have two directions, we need at least two vCPUs.
-        if mode == "bd" and basevm.vcpus_count < 2:
-            continue
-
-        for protocol in CONFIG_DICT["protocols"]:
-            # Distribute modes evenly between producers and consumers.
-            pipes_generator = create_pipes_generator(
-                basevm, mode, current_avail_cpu, protocol, host_ip, env_id
-            )
-
-            for cons, prod, pipe_tag in pipes_generator:
-                yield cons, prod, pipe_tag
+    iperf3_id = f"tcp-p{payload_length}-wsDEFAULT-{mode}"
+
+    cons = consumer.LambdaConsumer(
+        metadata_provider=DictMetadataProvider(
+            measurements=CONFIG_DICT["measurements"],
+            baseline_provider=NetTCPThroughputBaselineProvider(env_id, iperf3_id),
+        ),
+        func=consume_iperf_tcp_output,
+        func_kwargs={"vcpus_count": basevm.vcpus_count},
+    )
+
+    prod_kwargs = {
+        "guest_cmd_builder": iperf_guest_cmd_builder,
+        "basevm": basevm,
+        "current_avail_cpu": current_avail_cpu,
+        "runtime": RUNTIME_SEC,
+        "omit": WARMUP_SEC,
+        "load_factor": LOAD_FACTOR,
+        "modes": MODE_MAP[mode],
+    }
+    prod = producer.LambdaProducer(produce_iperf_output, prod_kwargs)
+    return cons, prod, f"{env_id}/{iperf3_id}"
 
 
 @pytest.mark.nonci
 @pytest.mark.timeout(3600)
 @pytest.mark.parametrize("vcpus", [1, 2])
+@pytest.mark.parametrize(
+    "payload_length", ["DEFAULT", "1024K"], ids=["pDEFAULT", "p1024K"]
+)
```

Comment:
Given [test: remove p=1024 case from vsock throughput perf test](https://github.com/firecracker-microvm/firecracker/pull/3736/commits/fdce3b8d8c6bbde994aa4bfe7402f06d11df2b4f) shouldn't this not contain `1024k`?

Reply:
So we used to test _three_ payload lengths: 1024, 1024**K** and DEFAULT. That commit only removes the first one